packages/dashboard-frontend/src/components/EditorTools/index.tsx

```diff
@@ -0,0 +1,160 @@
+};
+
+class EditorTools extends React.PureComponent<Props, State> {
+  private copiedTimer: number | undefined;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isExpanded: false,
+      contentText: '',
+      isDevWorkspace: false,
+      contentBlobUrl: '',
+    };
+  }
+
+  public componentDidMount(): void {
+    const { content } = this.props;
+    try {
+      const contentText = stringify(content);
+      const isDevWorkspace = this.isDevWorkspace(content);
+      const contentBlobUrl = URL.createObjectURL(
+        new Blob([contentText], { type: 'application/x-yaml' }),
+      );
+      this.setState({ contentText, isDevWorkspace, contentBlobUrl });
+    } catch (e) {
+      console.error(e);
+    }
+  }
+
+  public componentDidUpdate(): void {
+    const { content } = this.props;
+    try {
+      const contentText = stringify(content);
+      if (contentText !== this.state.contentText) {
+        const isDevWorkspace = this.isDevWorkspace(content);
+        const contentBlobUrl = URL.createObjectURL(
+          new Blob([contentText], { type: 'application/x-yaml' }),
+        );
+        this.setState({ contentText, isDevWorkspace, contentBlobUrl });
+      }
+    } catch (e) {
+      console.error(e);
+    }
+  }
+
+  private handleExpand(): void {
+    if (this.state.isExpanded) {
+      window.postMessage('show-navbar', '*');
+      const isExpanded = false;
+      this.setState({ isExpanded });
+      this.props.handleExpand(isExpanded);
+    } else {
+      window.postMessage('hide-navbar', '*');
+      const isExpanded = true;
+      this.setState({ isExpanded });
+      this.props.handleExpand(isExpanded);
+    }
+  }
+
+  private isDevWorkspace(
+    content: che.Workspace | devfileApi.DevWorkspace | che.WorkspaceDevfile | devfileApi.Devfile,
+  ): boolean {
+    if ((content as devfileApi.DevWorkspace)?.kind === 'DevWorkspace') {
+      return true;
+    }
+    if ((content as che.Workspace)?.devfile) {
+      return true;
+    }
+    return false;
+  }
+
+  private onCopyToClipboard(): void {
+    this.setState({ copied: true });
+    if (this.copiedTimer) {
+      clearTimeout(this.copiedTimer);
+    }
+    this.copiedTimer = window.setTimeout(() => {
+      this.setState({ copied: false });
+    }, 3000);
+  }
+
+  public render(): React.ReactElement {
+    const { contentText, isDevWorkspace, contentBlobUrl, isExpanded, copied } = this.state;
+
+    return (
+      <div className={styles.editorTools}>
+        <Flex>
+          <FlexItem>
+            <CopyToClipboard text={contentText} onCopy={() => this.onCopyToClipboard()}>
+              <Button variant="link">
+                <CopyIcon />
+                {copied ? 'Copied' : 'Copy to clipboard'}
+              </Button>
+            </CopyToClipboard>
+          </FlexItem>
+          <Divider isVertical />
+          <FlexItem>
+            <a
+              download={`${isDevWorkspace ? 'devworkspace' : 'devfile'}.yaml`}
```

Comment:
I'm thinking of adding a devworkspace (devfile) name in front, e.g., `my-project.devworkspace.yaml`. That should be more useful than simply `devworkspace.yaml`.
cc @ibuziuk